### ISSUE TITLE
How to update the suggestions list while typing in the searchbar?

### ISSUE THREAD

**Issue by al-t** [mancj/MaterialSearchBar#28]
I can't figure out how to update the suggestions list in the searchbar when typing in it. Could you help me please?

I have a TextWatcher on the searchbar and when the user types in the text the database is queried in the background, in AsyncTask, and then creates the list of suggestions. Then, it executes the method of my Activity which sets the suggestions to the searchbox.

Here is what that method does: 
```
public void updateSearch(List<Suggestion> suggestions) {
        customSuggestionsAdapter.setSuggestions(suggestions);
        searchBar.setCustomSuggestionAdapter(customSuggestionsAdapter);
}
```

But nothing happens in the searchbox. That is, the suggestion list does not appear. If I press the back button on the searchbar and disable the search and then enable it again, that is, I press on the search bar, then the suggestion list appears, so I gather it was actually set to the searhbar but did not refresh on the screen. But what should I do to make it appear and change while typing in the searchbox?
As a stupid hack I can get it around by automatic disable and enable of the seachBar right after I set the suggestion adapter, but that update doesn't look good visually, and I suppose I just missed something simple.

**Comment by mancj** [comment]
try the following

```
public void updateSearch(List<Suggestion> suggestions) {
        searchBar.updateLastSuggestions(suggestions);
}
```

**Comment by al-t** [comment]
Thank you! That helped. 

I actually tried it before asking here but the java.lang.ClassCastException was thrown, so I thought the updateLastSuggestions() method only expects List<String> as a parameter.
`com.example.app.search.Suggestion cannot be cast to java.lang.String
                                                                   at com.mancj.materialsearchbar.adapter.DefaultSuggestionsAdapter.onBindSuggestionHolder(DefaultSuggestionsAdapter.java:16)
`
That was because I didn't set the custom adapter before, so it was using the Default Suggestions Adapter, which expecsts suggestions to be Strings.
Now I set the custom suggestion adapter in the onCreate() method, where the searchbar is initialised, and everything works fine.
